### ISSUE TITLE
[Question] Working through multiprocess ftp downloads

### ISSUE THREAD

**Issue by ieaster1** [tqdm/tqdm#781]
This may actually be more apparent than I realize but working on a project while learning Python.  I'm currently down the rabbit hole to try and make this work so any help is appreciated.

I'm trying to have a progress bar for each package being downloaded through ftp using multiprocessing.  I assumed I had too many moving parts to understand it fully so I stripped it back to just loop through a list of packages from code I found in #129.

**Code Snippet:**
```
packages = (
    'package-01.tgz',
    'package-02.tgz',
    'package-03.tgz'
)

ftp = FTP('ftp_url')
ftp.login('user', 'pass')

for p in packages:
    with open(path + p, 'wb') as fd:
        total = ftp.size(p)
        with tqdm(total=total,
                  unit_scale=True,
                  desc=p,
                  miniters=1,
                  file=sys.stdout,
                  leave=True
                  ) as pbar:
            def cb(data):
                pbar.update(len(data))
                fd.write(data)
        ftp.retrbinary('RETR {}'.format(p), cb)
```

My specific issue here is that the progress bar never updates with downloaded chunks but is actively writing to disk.  It will loop through the list, download the packages, but the progress bar never actually updates.

```package-01.tgz:   0%|                                                                                       | 0.00/662M [00:00<?, ?it/s]```

**Comment by ieaster1** [comment]
Fell into the indentation pit.
`ftp.retrbinary('RETR {}'.format(p), cb)`
Needed to be indented to be part of the `with` block.
:man_facepalming:

Would I be able to simply pass the `position=n` argument to allow simultaneous progress bars with multiprocessing?  Currently, each progress bar instance is just updating on the same line.

**Comment by ieaster1** [comment]
To a noob like me, this felt fairly challenging.  I finally have things rolling to work as desired at this point.

For anyone else that potentially comes across this, my application generates packages to download to the system based on criteria.  Once I cleaned up my previous indentation issue, it worked well but the progress bars would rotate on the same line.  

I needed to pass a unique value to the position argument for each progressbar instance.

My solution was to just grab an index value from my list of packages and use that value as the position.

```
n = package_list.index('package_name')
```

Hope that makes sense to anyone else passing by.  I'll see my noob-self out now.  😄
